semver/range_test.ts

```diff
@@ -13,7 +13,6 @@ Deno.test("range", async (t) => {
     ["1.2.3-pre+asdf - 2.4.3-pre+asdf", "1.2.3"],
     ["1.2.3-pre+asdf - 2.4.3-pre+asdf", "1.2.3-pre.2"],
     ["1.2.3-pre+asdf - 2.4.3-pre+asdf", "2.4.3-alpha"],
-    ["1.2.3+asdf - 2.4.3+asdf", "2.4.3-alpha"],
```

Comment:
This case is incompatible `npm:semver`. The below prints false, but expected true here.

```js
import * as semver from "npm:semver";
const range = new semver.Range("1.2.3+asdf - 2.4.3+asdf");
console.log(range.test("2.4.3-alpha"));
```

Reply:
Is this test from `npm:semver` present in `std/semver` somewhere?

Reply:
I don't get the question well.. This case was tested with the opposite expectation (as `npm:semver`), and this PR removes it.

Do you suggest we should add this case with fixed expectation?

Reply:
> Do you suggest we should add this case with fixed expectation?

Yes. Ditto with the other similar comments.

Reply:
Added this case back to `testRange() returns false when the version is not in the range` with fixed expectation